runner/main.go

```diff
@@ -0,0 +1,269 @@
+package runner
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"regexp"
+	"runtime"
+	"strconv"
+	"testing"
+	"time"
+	"unsafe"
+)
+
+type (
+	testRunner struct {
+		m             *testing.M
+		intTests      *[]testing.InternalTest
+		intBenchmarks *[]testing.InternalBenchmark
+
+		tests      *map[string]*testDescriptor
+		benchmarks *map[string]*benchmarkDescriptor
+
+		failRetriesCount int
+		exitOnError      bool
+		exitCode         int
+	}
+	testDescriptor struct {
+		test    testing.InternalTest
+		fqn     string
+		ran     int
+		failed  bool
+		flaky   bool
+		error   bool
+		skipped bool
+	}
+	benchmarkDescriptor struct {
+		benchmark testing.InternalBenchmark
+		fqn       string
+		skipped   bool
+	}
+	internalTestResult struct {
+		ran        bool      // Test or benchmark (or one of its subtests) was executed.
+		failed     bool      // Test or benchmark has failed.
+		skipped    bool      // Test of benchmark has been skipped.
+		done       bool      // Test is finished and all subtests have completed.
+		finished   bool      // Test function has completed.
+		raceErrors int       // number of races detected during test
+		name       string    // Name of test or benchmark.
+		start      time.Time // Time test or benchmark started
+		duration   time.Duration
+	}
+)
+
+var runner *testRunner
+var runnerRegexName *regexp.Regexp
+
+// Runs a test suite
+func Run(m *testing.M, exitOnError bool, failRetriesCount int) int {
+	runner := &testRunner{
+		m:                m,
+		exitOnError:      exitOnError,
+		failRetriesCount: failRetriesCount,
+	}
+	runner.init()
+	return runner.Run()
+}
+
+// Gets the test name
+func GetTestName(name string) string {
+	if runnerRegexName == nil {
+		runnerRegexName = regexp.MustCompile(`(?m)([\w -:_]*)\/\[runner.[\w:]*](\/[\w -:_]*)?`)
+	}
+	match := runnerRegexName.FindStringSubmatch(name)
+	if match == nil || len(match) == 0 {
+		return name
+	}
+	return match[1] + match[2]
+}
+
+// Runs the test suite
+func (r *testRunner) Run() int {
+	r.exitCode = 0
+	r.m.Run()
+	return r.exitCode
+}
+
+// Internal test processor, each test calls the processor in order to handle retries, process exiting and exitcodes
+func (r *testRunner) testProcessor(t *testing.T) {
+	t.Helper()
+	name := t.Name()
+	if item, ok := (*r.tests)[name]; ok {
+
+		//Sets the original test name
```

Comment:
```suggestion
		// Sets the original test name
```
Be careful with the formatting of comments

Reply:
ok